core/domain/user_query_domain.py

```diff
@@ -21,22 +21,17 @@
 
 import collections
 
+from constants import constants
 import feconf
 import python_utils
 import utils
 
+__attribute_names__ = [
+    predicate['backend_attr'] for predicate in (
+        constants.EMAIL_DASHBOARD_PREDICATE_DEFINITION)]
 
-UserQueryParams = collections.namedtuple(
-    'UserQueryParams',
-    [
-        'inactive_in_last_n_days',
-        'has_not_logged_in_for_n_days',
-        'created_at_least_n_exps',
-        'created_fewer_than_n_exps',
-        'edited_at_least_n_exps',
-        'edited_fewer_than_n_exps'
-    ]
-)
+USER_QUERY_PARAMS = collections.namedtuple(
+    'USER_QUERY_PARAMS', __attribute_names__)
```

Comment:
Why not this? This is more like a class.

```suggestion
UserQueryParams = collections.namedtuple(
    'UserQueryParams', __attribute_names__)
```

Reply:
I changed it because of the linting checks in place.

Reply:
What was the specific lint failure that you encountered? This seems inconsistent with our other class name styles. I'd expect constants to follow UPPER_SNAKE_CASE, but that the type name itself stays consistent with other class names (i.e. UpperCamelCase).

Reply:
The lint error:
Constant name "UserQueryParams" doesn't conform to u'^(([A-Z_][A-Z0-9_]*)|(__.*__)|([a-z_]+_models)|([a-z_]+_services))$' pattern

Reply:
I think the detection is sometimes not working correctly, probably add an ignore here. Similar example https://github.com/oppia/oppia/pull/7351#discussion_r314307974

Reply:
I have added the lint ignore. Done.